magda-gateway/src/createOpenfaasGatewayProxy.ts

```diff
@@ -0,0 +1,32 @@
+import express from "express";
+import createBaseProxy from "./createBaseProxy";
+import { TenantMode } from "./setupTenantMode";
+import Authenticator from "./Authenticator";
+import { mustBeAdmin } from "magda-typescript-common/src/authorization-api/authMiddleware";
+
+interface OptionsType {
+    gatewayUrl: string;
+    baseAuthUrl: string;
+    jwtSecret: string;
+    authenticator: Authenticator;
+    tenantMode: TenantMode;
+    allowAdminOnly?: boolean;
+}
+
+export default function createOpenfaasGatewayProxy(
+    options: OptionsType
+): express.Router {
+    const router = express.Router();
```

Comment:
Could use a test for this - in particular to make sure that if anyone ever breaks this so that the `mustBeAdmin` thing stops working, it'll cause a build failure!

Reply:
will add a test case.

Reply:
Added test cases